WordPress/src/main/java/org/wordpress/android/ui/posts/CriticalPostActionTracker.kt

```diff
@@ -0,0 +1,43 @@
+package org.wordpress.android.ui.posts
+
+import org.wordpress.android.BuildConfig
+import org.wordpress.android.fluxc.model.LocalOrRemoteId.LocalId
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+
+class CriticalPostActionTracker(
+    // TODO: Find a better name than listener
+    private val listener: () -> Unit,
+    private val shouldCrashOnUnexpectedAction: Boolean = BuildConfig.DEBUG
+) {
+    enum class CriticalPostAction {
+        DELETING_POST, RESTORING_POST, TRASHING_POST
+    }
+    private val map = HashMap<LocalId, CriticalPostAction>()
+
+    fun add(localPostId: LocalId, criticalPostAction: CriticalPostAction) {
+        if (map.containsKey(localPostId)) {
+            val currentActionName = map[localPostId]?.name
+            val newActionName = criticalPostAction.name
+            val errorMessage = "We should not perform more than one critical post action. Current action is " +
+                    "($currentActionName), new action is ($newActionName)"
+            AppLog.e(T.POSTS, errorMessage)
+            if (shouldCrashOnUnexpectedAction) {
```

Comment:
I feel like this is one of the cases where we can end up in an invalid state so I'd personally crash the app even in the production. However, I'm okay if you want to keep it as is.

Reply:
I am not sure about this, this is only used as a flag for UI state. We want to catch these issues but not at the expense of the user. If we do use this component for something more, then I agree.

Reply:
Hmm, at the same time the ui state might theoretically prevent the user from taking any actions on the item (eg. if we keep showing the overlay) and I think we want to be sure we know about these issues even before the users contact our support. 
We might want to consider using `Crashlytics.log(..)` more instead of crashing the app in production for similar issues. The invalid state in this case doesn't have any persistent consequences (eg. invalid data in a database). Wdyt?

Reply:
That gives me an idea, how about we create a central place to handle these issues. We can pass in a block that resolves to `Boolean` or a simple `Boolean` (I am not sure which is better right away), a severity enum and a message and let that central place decide what to do with it. This will:

1. Bring consistency across every feature
2. Make it easy to change how we handle issues
3. (most importantly for me) Never have to think about how the issue will be handled except for assigning a severity to it.

---

Let's make this more severe, I am totally OK with that. I have been running countless tests on the branch and never had a crash after this PR. We can always revert that back before it hits the production if we find it unstable. Do you mind opening an issue for this as well? Thanks!

Reply:
I like the idea! I'll think about it. It might be difficult to come up with good names for the severities as what we do really depends on the particular error and it's potential consequences. But I like the idea of having a single place and easy to use component for everyone.

Reply:
Created an issue https://github.com/wordpress-mobile/WordPress-Android/issues/9605